### PR TITLE
Add header when building for other platforms

### PR DESCRIPTION
**Pull request by jasonsandlin**

### CODE REVIEW

**Review by amccalib — APPROVED** [review]

File: Include/httpClient/httpClient.h [hunk 1]
```diff
@@ -121,7 +121,9 @@ typedef struct HCInitArgs {
     /// <summary>The Java Application Context.</summary>
     jobject applicationContext;
 } HCInitArgs;
-#elif HC_PLATFORM != HC_PLATFORM_NINTENDO_SWITCH
+#elif HC_PLATFORM == HC_PLATFORM_NINTENDO_SWITCH
+#include <httpClient/httpClient_switch.h>
+#elif
```
Comment (jplafonta) [inline Include/httpClient/httpClient.h:126]: I'd prefer to just have the client include this header.  The goal was that the NDA'd platforms would be completely additive, not requiring changes to the primary repo as much as possible.

Comment (jasonsandlin) [reply Include/httpClient/httpClient.h:130]: changed design

File: Include/httpClient/httpClient.h [hunk 2]
```diff
@@ -111,7 +111,13 @@ STDAPI HCMemGetFunctions(
 
 struct HCInitArgs;
 
-#if HC_PLATFORM == HC_PLATFORM_ANDROID
+#ifndef LHC_PLATFORM_HEADER_OVERRIDE
```
Comment (lubeltra) [inline Include/httpClient/httpClient.h:114]: this should be in config.h

Comment (lubeltra) [reply Include/httpClient/httpClient.h:114]: oh and it should be HC_PLATFORM_HEADER_OVERRIDE not LCH_

**Review by lubeltra — APPROVED** [review]

**Review by lubeltra — APPROVED** [review]

File: Include/httpClient/httpClient.h [hunk 3]
```diff
@@ -111,7 +111,13 @@ STDAPI HCMemGetFunctions(
...
 struct HCInitArgs;
 
-#if HC_PLATFORM == HC_PLATFORM_ANDROID
+#if HC_PLATFORM_HEADER_OVERRIDE
+    #ifndef HC_PLATFORM_TYPES_PATH
+        #error HC_PLATFORM_TYPES_PATH is not defined.  Set to the path of the platform header to include.
+    #else
+        #include HC_PLATFORM_TYPES_PATH
+    #endif
+#elif HC_PLATFORM == HC_PLATFORM_ANDROID
```
Comment (jplafonta) [inline Include/httpClient/httpClient.h:120]: I'd move this to the top of the file alongside the rest of the includes

**Review by jplafonta — APPROVED** [review]
:shipit:

**Pull request merged**